### DISCUSSION TITLE
Disable selfBroadcast for Postgres realtime updates

### DISCUSSION THREAD

**Discussion by colinchadwick** [supabase/supabase#14955] [Questions] [answered]
Hey everyone! Is it possible to exclude updates oneself has made to a table when listening to realtime changes? Updates from all other users should be streamed as usual.

**Comment by GaryAustin1** [comment, marked as answer]
Assuming the you are talking postgres_changes part of realtime, no.   The realtime server does not know where the updates to tables are coming from.   Only way would be to have a filter to rule out the user's id if it is part of the data.  And you only get the one filter.  You could also do the same filter in client code if you have the user_id in the payload data.

**Reply by colinchadwick** [reply]
Thanks for letting me know! I think this feature would be fantastic because it would allow me to receive updates solely from other users. By doing so, I wouldn't have to depend on real-time updates, which can sometimes be unreliable, to update the client's state after they've made changes themselves.

**Reply by GaryAustin1** [reply]
I'm really not sure how the realtime server will ever know "who" made an update to a database table.  Unless you put the uuid of the user in a column in the table there is no information in the database about which user inserted/updated the data.   If you do have such a column then you could filter on it with the neq filter and user id in your realtime setup.

**Reply by colinchadwick** [reply]
I think having this is crucial for offering a good realtime experience. Firebase seems to have something similar: https://firebase.google.com/docs/firestore/query-data/listen#events-local-changes, which isn't possible in Supabase. Using `metadata.hasPendingWrites`, I can check whether the update is coming from the server or not.

**Reply by colinchadwick** [reply]
Please let me know in case Supabase already has something like this, but I don’t think so.

**Reply by colinchadwick** [reply]
I think I’ll use what you’ve said as a workaround. I’ll add a last_edited_by column to my table and send a session id each time a user updates the table. Relying on the UUID of a user could be problematic, as one could have multiple tabs open logged in as the same user.

**Reply by GaryAustin1** [reply]
Supabase does not have anything like that.  There is no user relationship to what you send in database calls to the realtime response.  They exist in different servers and the info passed from the database is just the row of data with only the columns in the row.   Realtime does not know what client did the update. 

Any matchup would have to be done on the client side and even then would be tricky, at least for updates, as two users could make the exact same change (assuming no user_id column) and there would be no way to tell the two updates apart.

**Reply by colinchadwick** [reply]
I've just implemented your suggestion (using a session id) and it's working great. Thanks for your help!
